### ISSUE TITLE
Project Overview/Manager 

### ISSUE THREAD

**Issue by Draylar** [natanfudge/Cloak#12]
Before we dive too deep into adding Fabric-specific features and utils, it would be good to discuss how the plugin manages and keeps track of a project as a whole. My primary focus would be on important files that we will need some form of access to, including:
  - fabric.mod.json
  - mixins.json
  - registry classes (Items/Blocks)
  - main and side initializers
  - base resource folder

So: an "abstraction later" (fancy language time) that allows for easy editing and watching of certain files. One use case would be for managing Lang files: call `Lang.addTranslation("en_us", "modid:thing", "My Thing")` to add a new translation entry to the English file. Obviously an over simplified example, but the point is there.

Another use case would be for tracking mod additions. This could provide the groundwork for having an overview list of *all* translations your mod provides, or something similar.

In other words, I'm saying we need a glorified mod addition tracking system that abstracts modding concepts.
